L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -0,0 +1,938 @@
+    for (auto& lpgbt_id : mapping_config_["Stage1"][stage1_id]["lpgbts"]) {
+      stage1_to_lpgbts_.emplace(stage1_id, lpgbt_id);
+    }
+  }
+
+  //lpgbt to Stage 1 mapping
+  for (unsigned lpgbt_id = 0; lpgbt_id < mapping_config_["lpgbt"].size(); lpgbt_id++) {
+    lpgbt_to_stage1_.emplace(lpgbt_id, mapping_config_["lpgbt"]["Stage1"]);
+  }
+
+  //lpgbt to module mapping
+  for (unsigned lpgbt_id = 0; lpgbt_id < mapping_config_["lpgbt"].size(); lpgbt_id++) {
+    for (auto& modules : mapping_config_["lpgbt"][lpgbt_id]["Modules"]) {
+      lpgbt_to_modules_.emplace(lpgbt_id, packLayerWaferId(modules["layer"], modules["u"], modules["v"]));
+    }
+  }
+
+  //module to lpgbt mapping
+  for (unsigned module = 0; module < mapping_config_["Module"].size(); module++) {
+    links_per_module_.emplace(packLayerWaferId(module["layer"], module["u"], module["v"]),
+                              mapping_config_["Module"][module]["lpgbts"].size());
+    for (auto& lpgbt : mapping_config_["Module"][module]["lpgbts"]) {
+      module_to_lpgbts_.emplace(packLayerWaferId(module["layer"], module["u"], module["v"]), lpgbt["id"]);
+    }
+  }
+
+  json_input_file.close();
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::packWaferId(int waferU, int waferV) const {
+  unsigned packed_value = 0;
+  unsigned waferUsign = (waferU >= 0) ? 0 : 1;
+  unsigned waferVsign = (waferV >= 0) ? 0 : 1;
+  packed_value |= ((std::abs(waferU) & HGCSiliconDetId::kHGCalWaferUMask) << HGCSiliconDetId::kHGCalWaferUOffset);
+  packed_value |= ((waferUsign & HGCSiliconDetId::kHGCalWaferUSignMask) << HGCSiliconDetId::kHGCalWaferUSignOffset);
+  packed_value |= ((std::abs(waferV) & HGCSiliconDetId::kHGCalWaferVMask) << HGCSiliconDetId::kHGCalWaferVOffset);
+  packed_value |= ((waferVsign & HGCSiliconDetId::kHGCalWaferVSignMask) << HGCSiliconDetId::kHGCalWaferVSignOffset);
+  return packed_value;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::packLayerWaferId(unsigned layer, int waferU, int waferV) const {
+  unsigned packed_value = 0;
+
+  packed_value |= ((waferU & HGCalModuleDetId::kHGCalModuleUMask) << HGCalModuleDetId::kHGCalModuleUOffset);
+  packed_value |= ((waferV & HGCalModuleDetId::kHGCalModuleVMask) << HGCalModuleDetId::kHGCalModuleVOffset);
+  packed_value |= ((layer & HGCalModuleDetId::kHGCalLayerMask) << HGCalModuleDetId::kHGCalLayerOffset);
+  return packed_value;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::packLayerModuleId(unsigned layer, unsigned module) const {
+  unsigned packed_value = 0;
+  unsigned subdet = ForwardSubdetector::HGCEE;
+  if (layer > heOffset_) {
+    layer -= heOffset_;
+    subdet = ForwardSubdetector::HGCHEF;
+  }
+  packed_value |= ((layer & HGCalDetId::kHGCalLayerMask) << HGCalDetId::kHGCalLayerOffset);
+  packed_value |= ((module & HGCalDetId::kHGCalWaferMask) << HGCalDetId::kHGCalWaferOffset);
+  packed_value |= ((subdet & DetId::kSubdetMask) << DetId::kSubdetOffset);
+  return packed_value;
+}
```

Comment:
Should not be needed since the id scheme used here doesn't correspond to the new module ids.

Reply:
Thanks, yes